src/StableTrie.mo

```diff
@@ -110,13 +112,17 @@ module {
       region.freeSpace -%= n;
     };
 
-    func newInternalNode(region : Region) : Nat64 {
+    func newInternalNode(region : Region) : ?Nat64 {
+      if (node_count == max_address) return null;
```

Comment:
This must be `if (node_count + 1 == max_address)` because we are incrementing `node_count` before returning it.

(node_count equals the number of inner nodes without the root node)